pinot-core/src/test/java/org/apache/pinot/core/accounting/WorkloadBudgetManagerTest.java

```diff
@@ -115,4 +115,31 @@ void testConcurrentTryChargeSingleWorkload() throws InterruptedException {
     assertEquals(initialMemBudget - totalMemCharged, remaining._memoryRemaining,
         "Memory budget mismatch after concurrent updates");
   }
+
+  @Test
+  void testCanAdmitQuery() {
```

Comment:
Are these tests enough to get good coverage of all scenarios / combinations ? How are we validating that ? 

Do we have a theoretical version / model of different permutations and combinations that we are using to test the functionality ?

Reply:
Have we written a unit or integration test to mimic the steady state pattern in PROD ?